src/NuGet.Clients/NuGet.CommandLine/Commands/SignCommand.cs

```diff
@@ -0,0 +1,211 @@
+        public string CertificateFingerprint { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCertificatePasswordDescription")]
+        public string CertificatePassword { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandCryptographicServiceProviderDescription")]
+        public string CryptographicServiceProvider { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandKeyContainerDescription")]
+        public string KeyContainer { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandHashAlgorithmDescription")]
+        public string HashAlgorithm { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandTimestamperDescription")]
+        public string Timestamper { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandTimestampHashAlgorithmDescription")]
+        public string TimestampHashAlgorithm { get; set; }
+
+        [Option(typeof(NuGetCommand), "SignCommandOverwriteDescription")]
+        public bool Overwrite { get; set; }
+
+        public override Task ExecuteCommandAsync()
+        {
+            var signArgs = GetSignArgs();
+            var signCommandRunner = new SignCommandRunner();
+            var result = signCommandRunner.ExecuteCommand(signArgs);
+
+            return Task.FromResult(result);
+        }
+
+        public SignArgs GetSignArgs()
+        {
+            ValidatePackagePath();
+            ValidateTimestamper();
+            ValidateCertificateInputs();
+            ValidateOutputDirectory();
+            ValidateAndParseStoreLocation(out var storeLocation);
+            ValidateAndParseStoreName(out var storeName);
+            ValidateAndParseHashAlgorithm(HashAlgorithm, nameof(HashAlgorithm), out var hashAlgorithm);
+            ValidateAndParseHashAlgorithm(TimestampHashAlgorithm, nameof(TimestampHashAlgorithm), out var timestampHashAlgorithm);
+
+            return new SignArgs()
+            {
+                PackagePath = Arguments[0],
+                OutputDirectory = OutputDirectory,
+                CertificatePath = CertificatePath,
+                CertificateStoreName = storeName,
+                CertificateStoreLocation = storeLocation,
+                CertificateSubjectName = CertificateSubjectName,
+                CertificateFingerprint = CertificateFingerprint,
+                CertificatePassword = CertificatePassword,
+                CryptographicServiceProvider = CryptographicServiceProvider,
+                KeyContainer = KeyContainer,
+                HashingAlgorithm = hashAlgorithm,
+                Logger = Console,
+                Overwrite = Overwrite,
+                NonInteractive = NonInteractive,
+                Timestamper = Timestamper,
+                TimestampHashAlgorithm = timestampHashAlgorithm
+            };
+        }
+
+        private void ValidateAndParseHashAlgorithm(string value, string name, out HashAlgorithmName hashAlgorithm)
```

Comment:
these methods can just return the value, it doesn't look like the out is needed since there is only 1 return